src/analyz/mod.rs

```diff
@@ -1088,7 +1126,26 @@ fn build_vtable_items<'tcx>(
     let trait_ref = tcx.instantiate_bound_regions_with_erased(trait_ref);
     let methods = tcx.vtable_entries(trait_ref);
 
-    let mut parts = Vec::with_capacity(methods.len());
+    // I'm not sure of the circumstances in which `methods` would or would not
+    // contain a `ty::vtable::VtblEntry::MetadataDropInPlace` entry, so add 1 to
+    // the allocation in case it doesn't.
+    let mut parts = Vec::with_capacity(methods.len() + 1);
```

Comment:
```suggestion
    // Reserve space for all methods plus the additional drop function.  Usually
    // `methods` will contain a few non-`VtblEntry::Method` entries, giving us extra
    // slots to work with, but we don't rely on that.
    let mut parts = Vec::with_capacity(methods.len() + 1);
```
I found the comment confusing (I had to check the loop body below to understand the connection to `VtblEntry::MetadataDropInPlace`) so I tried rewording it.